File: src/JavaImportComparer.cs
using System;
using System.Collections.Generic;

namespace AutoRest.Java
{
    public class JavaImportComparer : IComparer<string>
    {
        public int Compare(string lhsImport, string rhsImport)
        {
            int result;

            if (lhsImport == rhsImport)
            {
                result = 0;
            }
            else if (lhsImport == null)
            {
                result = -1;
            }
            else if (rhsImport == null)
            {
                result = 1;
            }
            else
            {
                result = 0;

                string[] lhsImportParts = GetImportParts(lhsImport);
                string[] rhsImportParts = GetImportParts(rhsImport);

                int minimumImportPartCount = Math.Min(lhsImportParts.Length, rhsImportParts.Length);
                for (int i = 0; i < minimumImportPartCount; ++i)
                {
                    string lhsImportPart = lhsImportParts[i].ToLowerInvariant();
                    string rhsImportPart = rhsImportParts[i].ToLowerInvariant();

                    if (!lhsImportPart.Equals(rhsImportPart))
                    {
                        bool isLastLhsPart = IsLastPart(i, lhsImportParts);
                        bool isLastRhsPart = IsLastPart(i, rhsImportParts);
                        if (isLastLhsPart != isLastRhsPart)
                        {
                            result = isLastLhsPart ? -1 : 1;
                        }
                        else
                        {
                            result = lhsImportPart.CompareTo(rhsImportPart);
                        }
                        break;
                    }
                }
            }

            return result;
        }

        private static string[] GetImportParts(string import)
        {
            return import.Split('.');
        }

        private static Boolean IsLastPart(int importIndex, string[] importParts)
        {
            return importIndex == importParts.Length - 1;
        }
    }
}

File: mapper/src/main/java/com/azure/autorest/model/javamodel/JavaImportComparer.java
package com.azure.autorest.model.javamodel;

import java.util.*;

public class JavaImportComparer implements Comparator<String>
{
    public final int compare(String lhsImport, String rhsImport)
    {
        int result;

        if (lhsImport.equals(rhsImport))
        {
            result = 0;
        }
        else if (lhsImport == null)
        {
            result = -1;
        }
        else if (rhsImport == null)
        {
            result = 1;
        }
        else
        {
            result = 0;

            String[] lhsImportParts = GetImportParts(lhsImport);
            String[] rhsImportParts = GetImportParts(rhsImport);

            int minimumImportPartCount = Math.min(lhsImportParts.length, rhsImportParts.length);
            for (int i = 0; i < minimumImportPartCount; ++i)
            {
                String lhsImportPart = lhsImportParts[i].toLowerCase(Locale.ROOT);
                String rhsImportPart = rhsImportParts[i].toLowerCase(Locale.ROOT);

                if (!lhsImportPart.equals(rhsImportPart))
                {
                    boolean isLastLhsPart = IsLastPart(i, lhsImportParts);
                    boolean isLastRhsPart = IsLastPart(i, rhsImportParts);
                    if (isLastLhsPart != isLastRhsPart)
                    {
                        result = isLastLhsPart ? -1 : 1;
                    }
                    else
                    {
                        result = lhsImportPart.compareTo(rhsImportPart);
                    }
                    break;
                }
            }
        }

        return result;
    }

    private static String[] GetImportParts(String import_Keyword)
    {
        return import_Keyword.split("[.]", -1);
    }

    private static boolean IsLastPart(int importIndex, String[] importParts)
    {
        return importIndex == importParts.length - 1;
    }
}